src/extra/libs/rlottie/lv_rlottie.c

```diff
@@ -52,233 +32,40 @@ static const char * path_create;
-}
-
-/**********************
- *   STATIC FUNCTIONS
- **********************/
-
-static void lv_rlottie_constructor(const lv_obj_class_t * class_p, lv_obj_t * obj)
-{
-    LV_UNUSED(class_p);
-    lv_rlottie_t * rlottie = (lv_rlottie_t *) obj;
-
-    if(rlottie_desc_create) {
-        rlottie->animation = lottie_animation_from_data(rlottie_desc_create, rlottie_desc_create, "");
-    }
-    else if(path_create) {
-        rlottie->animation = lottie_animation_from_file(path_create);
-    }
-    if(rlottie->animation == NULL) {
-        LV_LOG_WARN("The aniamtion can't be opened");
-        return;
-    }
-
-    rlottie->total_frames = lottie_animation_get_totalframe(rlottie->animation);
-    rlottie->framerate = (size_t)lottie_animation_get_framerate(rlottie->animation);
-    rlottie->current_frame = 0;
-
-    rlottie->scanline_width = create_width * LV_ARGB32 / 8;
-
-    size_t allocaled_buf_size = (create_width * create_height * LV_ARGB32 / 8);
-    rlottie->allocated_buf = lv_mem_alloc(allocaled_buf_size);
-    if(rlottie->allocated_buf != NULL) {
-        rlottie->allocated_buffer_size = allocaled_buf_size;
-        memset(rlottie->allocated_buf, 0, allocaled_buf_size);
-    }
-
-    rlottie->imgdsc.header.always_zero = 0;
-    rlottie->imgdsc.header.cf = LV_IMG_CF_TRUE_COLOR_ALPHA;
-    rlottie->imgdsc.header.h = create_height;
-    rlottie->imgdsc.header.w = create_width;
-    rlottie->imgdsc.data = (void *)rlottie->allocated_buf;
-    rlottie->imgdsc.data_size = allocaled_buf_size;
-
-    lv_img_set_src(obj, &rlottie->imgdsc);
-
-    rlottie->play_ctrl = LV_RLOTTIE_CTRL_FORWARD | LV_RLOTTIE_CTRL_PLAY | LV_RLOTTIE_CTRL_LOOP;
-    rlottie->dest_frame = rlottie->total_frames; /* invalid destination frame so it's possible to pause on frame 0 */
-
-    rlottie->task = lv_timer_create(next_frame_task_cb, 1000 / rlottie->framerate, obj);
-
-    lv_obj_update_layout(obj);
-}
-
-
-static void lv_rlottie_destructor(const lv_obj_class_t * class_p, lv_obj_t * obj)
-{
-    LV_UNUSED(class_p);
-    lv_rlottie_t * rlottie = (lv_rlottie_t *) obj;
-
-    if(rlottie->animation) {
-        lottie_animation_destroy(rlottie->animation);
-        rlottie->animation = 0;
-        rlottie->current_frame = 0;
-        rlottie->framerate = 0;
-        rlottie->scanline_width = 0;
-        rlottie->total_frames = 0;
-    }
-
-    if(rlottie->task) {
-        lv_timer_del(rlottie->task);
-        rlottie->task = NULL;
-        rlottie->play_ctrl = LV_RLOTTIE_CTRL_FORWARD;
-        rlottie->dest_frame = 0;
-    }
-
-    lv_img_cache_invalidate_src(&rlottie->imgdsc);
-    if(rlottie->allocated_buf) {
-        lv_mem_free(rlottie->allocated_buf);
-        rlottie->allocated_buf = NULL;
-        rlottie->allocated_buffer_size = 0;
-    }
-
-}
-
-#if LV_COLOR_DEPTH == 16
-static void convert_to_rgba5658(uint32_t * pix, const size_t width, const size_t height)
+lv_res_t lv_rlottie_from_file(lv_obj_t * obj, lv_coord_t width, lv_coord_t height, const char * path,
```

Comment:
Why can't we use the plain `lv_img` API?

Reply:
Yes, see above.